algorithms/kernel/classifier/classifier_train.cpp

```diff
@@ -114,6 +116,30 @@ services::Status Input::checkImpl(const daal::algorithms::Parameter *parameter)
     data_management::NumericTablePtr labelsTable = get(labels);
     DAAL_CHECK_STATUS(s, data_management::checkNumericTable(labelsTable.get(), labelsStr(), 0, 0, 1, nRows));
 
+    // Check correctness number of class
+    data_management::BlockDescriptor<int> yBD;
+    const_cast<data_management::NumericTable *>(labelsTable.get())->getBlockOfRows(0, nRows, data_management::readOnly, yBD);
+    const int * const dy = yBD.getBlockPtr();
+    const Parameter *algParameter = static_cast<const Parameter *>(parameter);
+    int nClasses{0};
+    if (INT_MAX > algParameter->nClasses)
+    {
+        nClasses = static_cast<int>(algParameter->nClasses);
+    }
+    else
+    {
+        DAAL_CHECK_STATUS(s, services::Status(services::ErrorIncorrectClassLabels));
+    }
+    for (size_t i = 0; i < nRows; ++i)
```

Comment:
I think it can be quite more faster if you will write smth like this:
```cpp
bool flag = false;
for (size_t i = 0; i < nRows; ++i)
{
     flag |= (dy[i] >= nClasses);
}
if (flag) return services::Status(services::ErrorIncorrectClassLabels);
```

Also, we can add threading here as well. It should eliminate possible perf regressions.

Reply:
agree, it was fixed